pkg/util/strategicpatch/patch_test.go

```diff
@@ -268,7 +284,111 @@ strategicMergePatchCases:
             - name: 2
               value: 2
         - name: 2
-sortMergeListTestCases:
+  - description: add field to map
+    original:
+      name: 1
+    patch:
+      value: 1
+    result:
+      name: 1
+      value: 1
+  - description: add field and delete field from map
+    original:
+      name: 1
+    patch:
+      name: null
+      value: 1
+    result:
+      value: 1
+  - description: delete field from nested map
+    original:
+      simpleMap:
+        key1: 1
+        key2: 1
+    patch:
+      simpleMap:
+        key2: null
+    result:
+      simpleMap:
+        key1: 1
+  - description: delete all fields from map
+    original:
+      name: 1
+      value: 1
+    patch:
+      name: null
+      value: null
+    result: {}
+  - description: add field and delete all fields from map
+    original:
+      name: 1
+      value: 1
+    patch:
+      other: a
+      name: null
+      value: null
+    result:
+      other: a
+`)
+
+// These are test cases for StrategicMergePatch that cannot be generated using
+// CreateStrategicMergePatch because it doesn't use the replace directive, generate
```

Comment:
doesn't use replace directive => uses the "replace" $patch directive instead of "delete"